### ISSUE TITLE
App crashes when use EpoxyPreloader and StaggeredGridLayoutManager

### ISSUE THREAD

**Issue by N3-M3-S1S** [airbnb/epoxy#1149]
When I use StaggeredGridLayoutManager and EpoxyPreloader my app crashes because there is an unchecked cast in EpoxyPreloader's onScrolled method at line 107: 

` val layoutManager = recyclerView.layoutManager as LinearLayoutManager`

There is no mention of non-linear layout managers in the Image Preloading wiki section. Is there any way to use EpoxyPreloader with StaggeredGridLayoutManager? 

Thanks in advance.

**Comment by elihart** [comment]
That's right. The LinearLayoutManager is used to calculate the visible item range so that we know where to start prefetching.

It seems like the same information is available from staggered layouts via functions like `findFirstVisibleItemPositions`, so it shouldn't be difficult to change the existing logic to be more generic and check the type of layout manager and use the relevant functions to calculate range. We could also create an interface as well to support custom layout managers.

Feel free to put up a PR for the change if you want it!
